content/en/blog/2021/migrate-alpha-security-policy/index.md

```diff
@@ -6,201 +6,177 @@ attribution: Yangmin Zhu (Google)
-body `Origin authentication failed`. In beta policy, the HTTP code 403 will be returned with the body `RBAC: access denied`.
+- a single `MeshPolicy` that applies to all services in the mesh;
+- a single namespace level `Policy` that applies to all workloads in the namespace;
+- multiple service level `Policy` that applies to the selected services in the namespace;
+- a single `ClusterRbacConfig` that enables the RBAC on the whole namespace or some services in the namespace;
+- multiple namespace level `ServiceRole` and `ServiceRoleBinding` that applies to all services in the namespace;
+- multiple namespace level `ServiceRole` and `ServiceRoleBinding` that applies to the selected services in the namespace;
 
-Additionally, the [`triggerRule.regex` field](https://istio.io/v1.4/docs/reference/config/security/istio.authentication.v1alpha1/#StringMatch)
-in the alpha JWT policy is not supported by the `AuthorizationPolicy`.
+### Step 2: Convert service name to label selector
 
-## `v1alpha1` authentication policy migration
+The `v1alpha1` policy selects target using service name, you should refer to the corresponding service definition to decide
+the label selector that should be used in the `v1beta1` policy.
 
-The typical flow of migrating to `v1beta1` policy is to start by checking all the `MeshPolicy` and `Policy` applied in the
-cluster and convert each of them to the corresponding `v1beta1` version.
+A single `v1alpha1` policy may include multiple services, this means it could be migrated to multiple `v1beta1` policies
+because the `v1beta1` policy currently only supports at most one label selector per policy.
 
-The `MeshPolicy` is mesh-level and there should be only 1 per-cluster, for each `MeshPolicy` and `Policy` applied in the cluster:
+Also note the `v1alpha1` uses service port but the `v1beta1` uses the workload port, this means the port number might be
+different in the migrated `v1beta1` policy.
 
-1. If the policy has service target, it is in service level. Find and take a note of the corresponding service definition.
-   You will use the service definition to convert the service target to the corresponding workload selector.
+### Step 3: Migrate authentication policy
 
-1. If the policy has no service target, it is in mesh/namespace level. The corresponding workload selector is just empty.
+For each `v1alpha1` authentication policy, migrate with the following rules:
 
-1. For each service target and mesh/namespace level policy, create an `PeerAuthentication` if the `peers` was used in
-   the `v1alpha1` policy. Populate the `selector` and `portLevelMtls` if it is a service level policy. Populate the
-   `mtls` mode with either `PERMISSIVE` or `STRICT` depending on the `v1alpha1` policy mode.
+1. If the whole namespace is enabled with mTLS or JWT, create the `PeerAuthentication`, `RequestAuthentication` and
+   `AuthorizationPolicy` with an empty label selector for the whole namespace. Fill out the policy based on the
```

Comment:
just say "without workload selector"? this wording makes me  thing its:
`selector: {}`